godot-core/src/obj/traits.rs

```diff
@@ -406,32 +417,32 @@ pub mod mem {
     pub struct DynamicRefCount {}
     impl Sealed for DynamicRefCount {}
     impl Memory for DynamicRefCount {
-        fn maybe_init_ref<T: GodotClass>(obj: &Gd<T>) {
+        fn maybe_init_ref<T: GodotClass>(obj: &RawGd<T>) {
             out!("  Dyn::init  <{}>", std::any::type_name::<T>());
             if obj
-                .instance_id_or_none()
+                .instance_id()
```

Comment:
Are these changes correct? `instance_id_or_none` performed a liveness check, but `instance_id` (to be renamed to `instance_id_unchecked`) does not.

Reply:
`maybe_init_ref` will call `RefCounted::init_ref` which goes through `out_class_ptrcall` which has:
```rs
if let Some(instance_id) = maybe_instance_id {
    crate::engine::ensure_object_alive(Some(instance_id), object_ptr, method_name);
}
```
So we still perform a liveness check here, just later than before.